drivers/display/gd7965.c

```diff
@@ -91,6 +83,49 @@ static inline int gd7965_write_cmd(const struct device *dev, uint8_t cmd,
 	return err;
 }
 
+static inline int gd7965_write_cmd_pattern(const struct device *dev,
+					   uint8_t cmd,
+					   uint8_t pattern, size_t len)
+{
+	const struct gd7965_config *config = dev->config;
+	struct spi_buf buf = {.buf = &cmd, .len = sizeof(cmd)};
+	struct spi_buf_set buf_set = {.buffers = &buf, .count = 1};
+	int err;
+	uint8_t data[64];
+
+	err = gpio_pin_set_dt(&config->dc_gpio, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	err = spi_write_dt(&config->bus, &buf_set);
+	if (err < 0) {
+		goto spi_out;
+	}
+
+	err = gpio_pin_set_dt(&config->dc_gpio, 0);
+	if (err < 0) {
+		goto spi_out;
+	}
+
+	memset(data, pattern, sizeof(data));
+	while (len) {
```

Comment:
IIRC I reworked ssd16xx.c to do in similar way. Also thought to add something like epd_common.c but still not sure it makes sense.

Reply:
I think that might be a bit tricky since the datasheets specify subtly different CS handling. The UltraChip-based ones suggest that CS should be toggled between data and command and ideally between every byte in the data.